godel_surface_detection/src/coordination/data_coordinator.cpp

```diff
@@ -364,5 +380,70 @@ namespace data
     ROS_WARN_STREAM(UNABLE_TO_FIND_RECORD_ERROR << " " << id);
     return false;
   }
+
+  /**
+   * @brief DataCoordinator::asyncSaveRecord calls saveRecord in a detached
+   * thread
+   * @param path directory of the save location
+   */
+  void DataCoordinator::asyncSaveRecord(boost::filesystem::path path)
+  {
+    try
+    {
+      auto thd = std::thread(&DataCoordinator::saveRecord, this, path);
+      thd.detach();
+    }
+    catch (const std::exception& e){
```

Comment:
terminate called after throwing an instance of 'pcl::IOException'
  what():  : [pcl::PCDWriter::writeASCII] Input point cloud has no data!